components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -542,18 +543,40 @@ function Import._groupLastVsAdditionalData(lpdbEntry)
 		table.insert(matchConditions, '[[match2id::' .. matchId .. ']]')
 	end
 
-	local matchData = mw.ext.LiquipediaDB.lpdb('match2', {
-		conditions = '[[opponent::' .. opponentName .. ']] AND (' .. table.concat(matchConditions, ' OR ') .. ')',
-		order = 'date desc, match2id desc',
-		query = 'date, match2opponents, winner',
-		limit = 1
-	})
+	local conditions = table.concat(matchConditions, ' OR ')
 
-	if not type(matchData) == 'table' or not matchData[1] then
+	if conditions ~= _last_vs_group_cache.conditions then
```

Comment:
Seems weird to use the condition as a cache hit indicator, why not directly check for a cache entry?
Otherwise this relies on the order of execution (i.e. the function has to called multiple times for the same opponent consecutively), or am i missing something?

Reply:
Ah, i missed that the condition isn't opponent-specific anymore, but is based on all matches of a group.